test/unit/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -245,26 +245,55 @@ void _nrn_mechanism_register_data_fields(Args&&... args) {
     extern "C" void __test_reg() {
         _initlists();
 
+        ion_reg("na", -10000.);
+
+        _na_sym = hoc_lookup("na_ion");
+
         register_mech(mechanism_info, nrn_alloc_pas_test, nrn_cur_pas_test, nrn_jacob_pas_test, nrn_state_pas_test, nrn_init_pas_test, hoc_nrnpointerindex, 1);
 
         mech_type = nrn_get_mechtype(mechanism_info[1]);
         _nrn_mechanism_register_data_fields(mech_type,
-            _nrn_mechanism_field<double>{"g"} /* 0 */,
-            _nrn_mechanism_field<double>{"e"} /* 1 */,
-            _nrn_mechanism_field<double>{"i"} /* 2 */,
-            _nrn_mechanism_field<double>{"ar", 2} /* 3 */,
-            _nrn_mechanism_field<double>{"s"} /* 4 */,
-            _nrn_mechanism_field<double>{"ena"} /* 5 */,
-            _nrn_mechanism_field<double>{"ina"} /* 6 */,
-            _nrn_mechanism_field<double>{"Ds"} /* 7 */,
-            _nrn_mechanism_field<double>{"v_unused"} /* 8 */,
-            _nrn_mechanism_field<double>{"g_unused"} /* 9 */
+            _nrn_mechanism_field<double>{"g"}, /* float var index 0 */
+            _nrn_mechanism_field<double>{"e"}, /* float var index 1 */
+            _nrn_mechanism_field<double>{"i"}, /* float var index 2 */
+            _nrn_mechanism_field<double>{"ar", 2}, /* float var index 3 */
+            _nrn_mechanism_field<double>{"s"}, /* float var index 4 */
+            _nrn_mechanism_field<double>{"ena"}, /* float var index 5 */
+            _nrn_mechanism_field<double>{"ina"}, /* float var index 6 */
+            _nrn_mechanism_field<double>{"Ds"}, /* float var index 7 */
+            _nrn_mechanism_field<double>{"v_unused"}, /* float var index 8 */
+            _nrn_mechanism_field<double>{"g_unused"}, /* float var index 9 */
+            _nrn_mechanism_field<double*>{"ion_ena", "ion_ena"}, /* int var index 0 */
```

Comment:
// note: coming to the test example it's bit clear! But mentioning this near code might be helpful?